crates/rpcsrv/src/session.rs

```diff
@@ -51,26 +50,50 @@ impl RemoteSession {
         let physical = session.create_physical_plan(plan).await?;
         let schema = physical.schema();
         let exec_id = session.add_physical_plan(physical)?;
-
-        Ok(PhysicalPlanResponse {
-            id: exec_id,
-            schema: Schema::clone(&schema),
-        })
+        Ok((exec_id, Schema::clone(&schema)))
     }
 
-    pub async fn dispatch_access(
-        &self,
-        table_ref: OwnedTableReference,
-    ) -> Result<TableProviderResponse> {
+    pub async fn dispatch_access(&self, table_ref: OwnedTableReference) -> Result<(Uuid, Schema)> {
         let mut session = self.session.lock().await;
         let provider = session.dispatch_access(table_ref).await?;
         let schema = provider.schema();
         let provider_id = session.add_table_provider(provider)?;
+        Ok((provider_id, Schema::clone(&schema)))
+    }
 
-        Ok(TableProviderResponse {
-            id: provider_id,
-            schema: Schema::clone(&schema),
-        })
+    pub async fn table_provider_scan(
+        &self,
+        provider_id: Uuid,
+        projection: Option<&Vec<usize>>,
+        filters: &[Expr],
+        limit: Option<usize>,
+    ) -> Result<(Uuid, Schema)> {
+        let mut session = self.session.lock().await;
+        let provider = session.get_table_provider(&provider_id)?;
+        // TODO: Use a context that actually matters.
+        let fake_ctx = SessionContext::new();
+        let physical = provider
+            .scan(&fake_ctx.state(), projection, filters, limit)
+            .await?;
+        let schema = physical.schema();
+        let exec_id = session.add_physical_plan(physical)?;
+        Ok((exec_id, Schema::clone(&schema)))
+    }
```

Comment:
Is scanning here actually needed? Since we're passing in logical plans, those will get converted into physical plans on the remote side, which is what's actually doing the scanning.

As an aside, I think we'll want to convert everything into an execution plan so that we're only dealing with logical and physical plans. Currently we have things like `DropTable` logical plans that we execute directly. If we instead do `DropTable` -> `DropTableExec`, we don't have to special case on the logical plans.

Reply:
Yeah, it's not required. Not sure it won't be when we have hybrid execution, though. I am not sure how we're going to modify the query planner. Keeping it here, we can remove it later if not required.